apps/meteor/tests/unit/app/lib/server/lib/notifyListener.spec.ts

```diff
@@ -0,0 +1,215 @@
+                useRealName: false,
+                expectedResult: undefined,
+            },
+            {
+                description: 'should hide message if type is user-unmuted',
+                message: { ...sampleMessage, t: 'user-unmuted' },
+                hideSystemMessages: ['user-unmuted'],
+                useRealName: false,
+                expectedResult: undefined,
+            },
+            {
+                description: 'should hide message if type is subscription-role-added',
+                message: { ...sampleMessage, t: 'subscription-role-added' },
+                hideSystemMessages: ['subscription-role-added'],
+                useRealName: false,
+                expectedResult: undefined,
+            },
+            {
+                description: 'should hide message if type is message_pinned',
+                message: { ...sampleMessage, t: 'message_pinned' },
+                hideSystemMessages: ['message_pinned'],
+                useRealName: false,
+                expectedResult: undefined,
+            },
+            {
+                description: 'should hide message if type is new-owner',
+                message: { ...sampleMessage, t: 'new-owner' },
+                hideSystemMessages: ['new-owner'],
+                useRealName: false,
+                expectedResult: undefined,
+            },
+        ];
+
+        testCases.forEach(({ description, message, hideSystemMessages, useRealName, expectedResult }) => {
+            it(description, async () => {
+                messagesFindOneStub.resolves(message);
+                getSettingValueByIdStub.withArgs('Hide_System_Messages').resolves(hideSystemMessages);
+                getSettingValueByIdStub.withArgs('UI_Use_Real_Name').resolves(useRealName);
+
+                if (useRealName) {
+                    usersFindOneStub.resolves({ name: 'Real User' });
+                }
+
+                const result = await getMessageToBroadcast({ id: '123' });
+
+                expect(result).to.deep.equal(expectedResult);
+            });
+        });
+    });
+
+    describe('notifyOnMessageChange', () => {
+        const setupProxyMock = (dbWatchersDisabled: boolean) => {
+            const proxyMock = proxyquire.noCallThru().load('../../../../../../app/lib/server/lib/notifyListener', {
+                '@rocket.chat/models': modelsStubs(),
+                '@rocket.chat/core-services': coreStubs(dbWatchersDisabled),
+                'mem': memStub,
+            });
+            notifyOnMessageChange = proxyMock.notifyOnMessageChange;
+        };
+
+        const testCases = [
+            {
+                description: 'should broadcast the message if dbWatchersDisabled is true',
+                dbWatchersDisabled: true,
+                expectBroadcast: true,
+            },
+            {
+                description: 'should not broadcast the message if dbWatchersDisabled is false',
```

Comment:
What about a new case when there is no message to broadcast? Does it make sense?